pkgs/development/libraries/libiconv/setup-hook.sh

```diff
@@ -0,0 +1,7 @@
+# libiconv must be listed in load flags on non-Glibc
+# it doesn't hurt to have it in Glibc either though
+iconvLdflags() {
+    export NIX_LDFLAGS="$NIX_LDFLAGS -liconv"
```

Comment:
Shouldn't we ensure that we don't keep adding this multiple times? e.g. check that `-liconv` is not already in `$NIX_LDFLAGS` before rexporting?

Reply:
We could but even then we wouldn't know if the package itself passes -liconv. I think it's safe to pass it twice to gcc and clang. I don't get any warnings at least.

Reply:
Right, multiple `-l` flags are usually no issues, and might even be preferable with cyclic dependencies.

I was just confused as it ended up four times in my ghc derivation. And I got four error messages. At first I though it would have to be in four different places.

```
env-vars
43:declare -x NIX_TARGET_LDFLAGS=" -L/nix/store/0pnvlr2ghaigvlwn2zg4nialw3sx5xmn-libiconv-1.15-x86_64-pc-mingw32/lib -liconv -liconv -L/nix/store/0pnvlr2ghaigvlwn2zg4nialw3sx5xmn-libiconv-1.15-x86_64-pc-mingw32/lib -liconv -liconv"

ghc-d0d02e2/libffi/build/x86_64-pc-mingw32/libtool
11829:postdeps="-liconv -liconv -liconv -liconv -lmingw32 -lgcc_s -lgcc -lmoldname -lmingwex -lmsvcrt -ladvapi32 -lshell32 -luser32 -lkernel32 -lmingw32 -lgcc_s -lgcc -lmoldname -lmingwex -lmsvcrt"
```

Reply:
Yeah that's probably package propagating libiconv incorrectly. There are quite a few that do this still & probably need to be fixed. I think that's where the duplicates are coming from.

https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=propagatedbuildinputs+libiconv&type=

Reply:
Did this get cross support? I thought it did.

Reply:
Hopefully- I'm not sure if anyone has used it yet in cross compile. In theory at least though.